test/units/TEST-58-REPART.sh

```diff
@@ -1685,6 +1685,72 @@ EOF
     grep -q 'UUID=[0-9a-f-]* /home btrfs discard,rw,nodev,suid,exec,subvol=@home,zstd:1,noatime,lazytime 0 1' "$root"/etc/fstab
 }
 
+testcase_btrfs_compression() {
+    local defs imgs loop output
+
+    if ! command -v btrfs &>/dev/null; then
+        echo "btrfs-progs not installed, skipping btrfs compression test."
+        return 0
+    fi
+
+    defs="$(mktemp -d)"
+    imgs="$(mktemp -d)"
+    # shellcheck disable=SC2064
+    trap "rm -rf '$defs' '$imgs'" RETURN
+    chmod 0755 "$defs"
+
+    echo "*** testcase for btrfs compression with CopyFiles (OFFLINE=$OFFLINE) ***"
+
+    dd if=/dev/zero of="$imgs/test-source-file" bs=1M count=1 2>/dev/null
+    test -f "$imgs/test-source-file"
```

Comment:
CopyFiles= cannot be in tmpfs or APIVFS (i.e. /proc/, /sys/, /dev/, /tmp/, /run/, /var/tmp/).
https://github.com/systemd/systemd/blob/c29e304562819f54c48baaaaaf98f98ee8d9d3ac/src/repart/repart.c#L5879-L5880
So, you get
```
[   20.126893] H TEST-58-REPART.sh[749]: /tmp/tmp.iZ2njS9EZF/test-source-file is in the denylist, ignoring
```

How about to copy files that always exist, e.g. /etc/systemd/system.conf or /etc/machine-id or so?

Reply:
aha! thank you, i had missed that.